### ISSUE TITLE
Incorrect OCI Mediatype after copy from containers-storage

### ISSUE THREAD

**Issue by besnardjb** [containers/skopeo#548]
Dear maintainers,

> tl;dr:
> I end up with an incorrect OCI mediaType in the final image ("application/vnd.oci.image.layer.v1.tar" instead of "application/vnd.oci.image.layer.v1.tar+gzip").
> 

I would like to report an oddity I've encountered while using skopeo in conjunction with buildah. Given my limited knowledge of the codebase, it appeared to me that the issue was related to the *skopeo copy* operation, I hope that it is correct.

This is how I encounter the issue:

buildah @ https://github.com/projectatomic/buildah/commit/08c56308c53c6525ef9060acde2c83026c5186eb
skopeo @ 82186b916faa9c8c70cfa922229bafe5ae024dec

```
# cat ~/Dockerfile 
from alpine
run apk update
run apk add vim
# buildah bud -t myalpine ~
(...)
# skopeo copy containers-storage:localhost/myalpine oci:myalpine-oci
(...)
# cat myalpine-oci/blobs/sha256/7f1e9b82c40554ac10736301440c3c039cea2c75d701b05a4042e622c672a0ff | jq 
{
  "schemaVersion": 2,
  "config": {
    "mediaType": "application/vnd.oci.image.config.v1+json",
    "digest": "sha256:0d67e56c33b13724414ebcacd937defd6d3554af21aa565865e525e90b6310b4",
    "size": 721
  },
  "layers": [
    {
      "mediaType": "application/vnd.oci.image.layer.v1.tar",
      "digest": "sha256:8e3ba11ec2a2b39ab372c60c16b421536e50e5ce64a0bc81765c2e38381bcff6",
      "size": 2206542
    },
    {
      "mediaType": "application/vnd.oci.image.layer.v1.tar+gzip",
      "digest": "sha256:73227f01ebef2b94007ae82318337546720979eaee2871648e7cf147c7b8cad8",
      "size": 10358984
    }
  ]
}
# file myalpine-oci/blobs/sha256/8e3ba11ec2a2b39ab372c60c16b421536e50e5ce64a0bc81765c2e38381bcff6
myalpine-oci/blobs/sha256/8e3ba11ec2a2b39ab372c60c16b421536e50e5ce64a0bc81765c2e38381bcff6: gzip compressed data

```

We can see that blob *8e3ba11ec2a2b39ab372c60c16b421536e50e5ce64a0bc81765c2e38381bcff6* is tagged "application/vnd.oci.image.layer.v1.tar" but practically gzipped. This may confuse a bundle extractor trusting OCI layer meta-data (e.g. umoci here https://github.com/openSUSE/umoci/blob/86a1af2e89e98bfc2327ac3d0655ab4b30554cc2/oci/layer/unpack.go#L88).

From my (limited) understanding, it seems that the resulting layer type was set when pulling the image outside of the containers-storage in the following file *containers/storage/storage_image.go:192* through the following line : 

```
uncompressedLayerType = imgspecv1.MediaTypeImageLayer
```

It seems that it was not set back to *MediaTypeImageLayerGzip* afterwards.

Thanks!

**Comment by mtrmac** [comment]
Thanks for the report.

Yes, this is a known bug: `copy.Image`, in `copyBlobFromStream`, can compress/decompress layers, but does not adjust MIME types.

**Comment by rhatdan** [comment]
@mtrmac @vrothberg Is this something we are going to fix?

**Comment by mtrmac** [comment]
Hopefully.

**Comment by mtrmac** [comment]
This might have been fixed by https://github.com/containers/image/pull/563 .

**Comment by besnardjb** [comment]
I confirm this problem is fixed in current master @ f72e39fc1073441ffc92aca2b46c72f7a5668fc3.

I repeated the steps of my small reproducer, with a previously installed version version 0.1.36-dev commit: 854f766dc7276db4fdbc4c3380b59c67c83cc3d0:

```json
    {
      "mediaType": "application/vnd.oci.image.layer.v1.tar",
      "digest": "sha256:519cb705bc6d8fc480c8ca57488bcc9f588997e3030978ba4b8b47030831aa67",
      "size": 2861852
    },
```

```bash
$ file blobs/sha256/519cb705bc6d8fc480c8ca57488bcc9f588997e3030978ba4b8b47030831aa67
blobs/sha256/519cb705bc6d8fc480c8ca57488bcc9f588997e3030978ba4b8b47030831aa67: gzip compressed data
```

And now with version 0.1.40-dev commit: f72e39fc1073441ffc92aca2b46c72f7a5668fc3

```json
    {
      "mediaType": "application/vnd.oci.image.layer.v1.tar+gzip",
      "digest": "sha256:519cb705bc6d8fc480c8ca57488bcc9f588997e3030978ba4b8b47030831aa67",
      "size": 2861852
    },
```

Thanks !

**Comment by vrothberg** [comment]
Thanks for reporting and checking, @besnardjb!
